criteria.go

```diff
@@ -156,6 +156,33 @@ func (f *field) In(values ...interface{}) *Criteria {
 	}
 }
 
+func (f *field) Contains(elems ...interface{}) *Criteria {
+	return &Criteria{
+		p: func(doc *Document) bool {
+			docValue := doc.Get(f.name)
+			if docValue == nil || reflect.TypeOf(docValue).Kind() != reflect.Slice {
+				return false
+			}
+			rv := reflect.ValueOf(docValue)
+			len := rv.Len()
+			set := make(map[interface{}]int)
```

Comment:
Use a `map[interface{}]bool` instead

Reply:
I use an `int` value to count duplicate elements. If we use a bool, then `Contains` criteria will pass for say,

```
Contains(4,4,2,4,4)
```

```
{
  "myField": [4,2]
}
```

To my understanding, this should return false, because 4,4,2,4,4 is not a subset of 4,2.

Reply:
The `Contains()` function should simply, for each element, if it is contained in the slice field. 
According to this description, it is not supposed to count elements

Reply:
Okay if that is the case, then perhaps the quadratic runtime solution I provided in the description will be better? I read that slices are not hashable map keys, so you were right to be concerned about that.

Reply:
Yes, I think it is better for starting.
An advanced version would mix the two algorithms, running the quadratic algorithm only for slices, maps or structs.
But for now, it is fine to implement the quadratic one in all cases